apps/src/code-studio/components/AudioRecorder.jsx

```diff
@@ -61,21 +62,34 @@ export default class AudioRecorder extends React.Component {
   };
 
   saveAudio = (blob) => {
-    assetsApi.putAsset(this.state.audioName + ".mp3", blob,
-    (xhr) => {
-      this.setState({audioName: ""});
-      this.props.onUploadDone(JSON.parse(xhr.response));
-      this.props.afterAudioSaved(AudioErrorType.NONE);
-    }, error => {
-      console.error(`Audio Failed to Save: ${error}`);
-      this.props.afterAudioSaved(AudioErrorType.SAVE);
-    });
+    if (!this.state.cancelling) {
```

Comment:
I'd move this check up a level from `saveAudio` into your recorder's `onstop` handler, for two reasons:

1. It'll wrap fewer lines, making the logic easier to read.
2. You can also bypass the creation of a `Blob` from the audio slices, saving a little bit more work.

```js
this.recorder.onstop = () => {
  if (!this.canceling) {
    const blob = new Blob(this.slices, {type: 'audio/mpeg'});
    this.saveAudio(blob);
  }
  this.slices = [];
};
```